Text input fields

cfarm opened the pull request:
Updates text input styles for form updates in [Design Manual Issue 346](https://github.com/cfpb/design-manual/issues/346)
## Additions
- `@input-disabled` and `@input-disabled-bg` color variables
- `.disabled` input style
## Changes
- `@input-error` updated hex value to match new standard
## Review
- @sonnakim
- @Scotchester 
## Screenshots

Inputs in the Design Manual
![screen shot 2015-09-24 at 4 05 13 pm](https://cloud.githubusercontent.com/assets/702526/10085378/e0403bce-62d6-11e5-8378-43ec19e2bad2.png)
## Notes
- related default input styles changed in [cf-core PR 46](https://github.com/cfpb/cf-core/pull/46)
- See related change to color variables in [Capital Framework PR 184](https://github.com/cfpb/capital-framework/pull/184) and [generator-cf PR 95](https://github.com/cfpb/generator-cf/pull/95)
## Checklist
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [ ] Passes all existing automated tests
- [ ] New functions include new tests
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged
- [x] Visually tested in supported browsers and devices 
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

cfarm committed:
initial disabled state styles; removed success input styles

cfarm committed:
disabled input bg color

cfarm committed:
update repo url

cfarm committed:
remove success text input icon

cfarm committed:
remove input icons

cfarm committed:
remove .warning styles, update error red color hex value

cfarm committed:
revert bower package

cfarm committed:
simplify disabled styles

cfarm committed:
remove deleted var from topdoc

Scotchester commented on src/cf-forms.less:
```diff
@@ -137,87 +119,14 @@ textarea {
-        outline: 1px solid @input-success;
+    &.disabled {
+        outline: 1px solid @input-disabled;
+        background-color: @input-disabled-bg;
```
Where is `@input-disabled-bg` being defined?

Scotchester commented:
Where will the field-level error message styling be implemented?

cfarm committed:
fix input-disabled var and disabled style

cfarm committed:
update default disabled color

cfarm committed:
undelete icons

cfarm commented:
I restored the icon styles, since I'll be using the error icon in the updated validation styles (in upcoming PR)

cfarm committed:
restore success var

sonnakim commented:
@Scotchester @cfarm as we (separately) discussed -
- We're keeping field-level warning and field-level success styles in the code base, even though they're not documented in this latest iteration of the Design Manual. (There are a few live pages that currently use both styles and we don't want them to break.)
- We _may_ add field-level success styles back to this latest DM iteration as there may be a good use case for them. But we will decided on that at a later point. 
- We're going to keep field-level warning styles out of the Design Manual  

Also, page-level warning styles are documented in the Alerts and Error Messaging DM page, not in the Form Fields page.

jimmynotjim commented:
Is this correct? 
<img width="393" alt="screen shot 2015-09-25 at 1 51 47 pm" src="https://cloud.githubusercontent.com/assets/1280430/10108028/a0d66484-638c-11e5-8b98-f590d50eabcc.png">

That disabled input looks like it's still a 2px border

Scotchester commented on src/cf-forms.less:
```diff
@@ -137,17 +122,12 @@ textarea {
-    &.success {
-        border: 1px solid @input-success;
-        outline: 1px solid @input-success;
+    &.disabled {
```
Missing closing brace for `&.success` declaration.

Scotchester commented on src/cf-forms.less:
```diff
@@ -137,17 +122,12 @@ textarea {
     }
 }
 
-
```
Please leave two blank lines between each Topdoc block.

Scotchester commented on src/cf-forms.less:
```diff
@@ -216,7 +196,6 @@ textarea {
     color: @input-success;
 }
 
-
```
Please leave two blank lines between each Topdoc block.

cfarm committed:
restore warning & success styles

cfarm committed:
topdoc spacing

cfarm committed:
error color var reverted

cfarm commented:
### Updates
- Restored icon styles
- Restored .warning and .success field level styles
- Updated error color to non-brand variable since we have the brand styling in cf-theme-overrides.less
- Fixed spacing

cfarm committed:
make disabled text input border 1px

cfarm commented:
Updated .disabled input border to be 1px.

cfarm committed:
update docs + demo

Scotchester commented on src/cf-forms.less:
```diff
@@ -452,4 +461,4 @@ textarea {
   name: EOF
   eof: true
-*/
+*/
```
Please restore EOF blank line

Scotchester commented on src/cf-forms.less:
```diff
@@ -100,9 +104,6 @@
         - .error
-      notes:
-        - "See the 'Form icons' section below for guidance on adding icons to
-           states."
```
This comment should be restored, yes?

jimmynotjim commented:
:+1: after addressing Scott's last two minor comments.

cfarm committed:
whitespace and comment fixes

cfarm commented:
moar fixes added.

Scotchester commented:
Kind of amusing that all this PR amounted to in the end was adding the disabled state.

cfarm commented:
![image](https://cloud.githubusercontent.com/assets/702526/10398769/d9a68a3c-6e7c-11e5-9b6c-9d1db13a0abf.png)
